### DISCUSSION TITLE
Laser-solid interaction

### DISCUSSION THREAD

**Discussion by berceanu** [AMReX-Astro/Castro#2568] [Ideas]
Hi all,

Can `Castro` be used to model the plasma formed by the interaction of a high power, ultra-short laser with a solid target of given geometry and atomic composition? 

If so, what would be a good starting point? Also, what are some of the main differences wrt `FLASH` for this use-case?

Thanks!
Andrei

**Comment by zingale** [comment]
Castro and Flash use similar algorithms.  The main difference is the AMR and time-integration strategies.  For your problem, how radiation is treated may also be important.  Do you need to include the effects of radiation?  or do you wish to start with just a lot of energy deposited in a region and then watch the interaction.

We don't support complex geometries.  But it sounds like you would have a target of one composition inside the domain of a different composition?  Is that correct?

**Reply by berceanu** [reply]
By radiation here do you mean laser radiation or secondary emission from the target? To get a better idea of the physics involved, a common setup is a structured nanotube target (with or without a substrate) placed in vacuum and irradiated by a high power laser. 
<img width="343" alt="Screenshot 2023-09-21 at 16 46 03" src="https://github.com/AMReX-Astro/Castro/assets/4405658/f81c4fc3-85d3-4e2d-82f6-5aa78603d718">

The laser intensity can go up to 10^17 W/cm^2 and the irradiation time up to hundreds of picoseconds.

**Reply by maximumcats** [reply]
He means do you need to directly model the laser radiation field in the simulation, or can it be indirectly modelled as a source term on the energy/temperature of the target without an explicit radiation field?

**Reply by berceanu** [reply]
We need to directly model it, for example in `FLASH` it is modeled using a ray tracing approach iirc.

https://flash.rochester.edu/site/flashcode/user_support/tutorial_talks/RAL_May2012/lasermgd.pdf

**Comment by zingale** [comment]
we do not have such a module, but would welcome a contribution.
